### ISSUE TITLE
AuthzClient fails if it can not access the frontend-url

### ISSUE THREAD

**Issue by QumeaAP** [keycloak/keycloak#12549]
### Describe the bug

When using the AuthzClient in a backend server, which can only access the keycloak through the backend-url not the frontend-url fails. 

The problem is, that the .well-known/uma2-configuration returns the frontend-urls for the "policy_endpoint" and the "permission_endpoint".

We run into the problem when using quarkus integration tests, where the keycloak is run inside the same docker-network as the docker-container under test. So the quarkus-application can access the keycloak only through the backend-url, whereas the application running the unit-tests can only access through the front url, the port bound to localhost.

There should either be a possibility to get the uma2-configuration with backend-urls. Possibly return backend-url endpoints when accessing through the backend-url and return frontend-url when accessing through the frontend-url?

Or there should be a possibility to override the urls for the AuthzClient instead of using the well-known endpoint.

### Version

18.0.0

### Expected behavior

The AuthzClient should also be working with backend-urls only

### Actual behavior

The AuthzClient fails if it has no access to the frontend-urls

### How to Reproduce?

* Configur a frontendurl for the 
* Call the https://keyclaok-backend-url/realms/{relam}/.well-known/uma2-configuration., then you get the "policy_endpoint" and as a frontent-url.

**Comment by pedroigor** [comment]
@QumeaAP You're right that the endpoint should be based on the backend URL as well as others like `permission_endpoint` and `resource_registration_endpoint`.

**Comment by pedroigor** [comment]
@QumeaAP Sent a fix, I appreciate if you could review it. See https://github.com/keycloak/keycloak/pull/12561.
